File: 0062-unique-paths/0062-unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
//         vector<vector<int>> dp(m, vector<int> (n, 0));
        
//         for(int i=1; i<m; i++){
//             for(int j=1; j<n; j++){
//                 dp[i][j] = dp[i-1][j] + dp[i][j-1] + 1; 
//             }
//         }
        
        long ans = 1;
        
        for(int i=m+n-2, j=1; i>=max(m, n); i--, j++){
            ans = (ans*i)/j;
        }
        
        
        return ans;    
    }
};

File: 0062-unique-paths/0062-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        grid = [[-1 for _ in range(n)] for _ in range(m)]
        
        grid[0][0] = 1
        
        for i in range(m):
            for j in range(n):
                if i == j == 0:
                    continue
                elif i == 0 or j == 0:
                    grid[i][j] = 1
                else:
                    grid[i][j] = grid[i-1][j] + grid[i][j-1]
        
        
        
        return grid[m-1][n-1]